### ISSUE TITLE
Add Hexagonal Isometric projection

### ISSUE THREAD

**Issue by bjorn** [bjorn/tiled#1508]
Currently Tiled supports Hexagonal maps with offset coordinates. There are [other possible coordinate systems](http://www.redblobgames.com/grids/hexagons/#coordinates), and one of them is Axial coordinates. This can actually be used already in Tiled using the Isometric projection, as [illustrated by Galdred on the forum](http://discourse.mapeditor.org/t/hexagon-coordinates/1934/3):

![eb9adb437a2e4d7da9b1553d2082848c033a063e](https://cloud.githubusercontent.com/assets/531764/24219691/fc349158-0f47-11e7-9268-d1a907cdfc20.png)

This issue is about adding a new "Hexagonal (Isometric)" or "Hexagonal (Axial)" (not sure about the best name) projection that derives from the `IsometricRenderer` and performs things like grid rendering and mouse picking correctly for hexagonal tiles.

Hopefully we can re-use the existing hexagonal map related properties, though their names would be a bit confusing (whether the hexagons are flat topped or pointy topped depends on the "stagger axis" property, and there is no staggering axis when using axial coordinates). The property could be renamed to "Hexagon Orientation" with values being either Horizontal / Vertical or Flat Topped / Pointy Topped.

**Comment by eishiya** [comment]
Popping in to say that renaming the stagger axis property to "Hexagon orientation" would make it easier to understand in general in my opinion, even if this projection isn't added. For the values, I think "Flat Top"/"Pointy Top" would be easy to understand, and concise.

**Comment by bjorn** [comment]
@eishiya True, though the stagger axis property also applies to staggered isometric maps, where "flat top" and "pointy top" make no sense.

**Comment by eishiya** [comment]
Aye, but I think it would be nice to see some intuitive naming, even if
it's the same property under the hood.
